examples/factors/README.md

```diff
@@ -1,22 +1,213 @@
+    if a == 1 || b == 1 {
+        panic!("Trivial factors")
+    }
+```
+
+## Compute and publish the product
+
+Now we can compute their product and `commit` it.
+```rust
+    let product = a.checked_mul(b).expect("Integer overflow");
+    env::commit(&product);
+```
+The `env::commit` function is used to commit public results to the [journal].
+Once committed to the journal, anyone with the [receipt] can read this value.
+
+Now you should be able to confirm your work again with `cargo run --release`.
+Once more, the program won't do anything, but it should run successfully and build with a single warning about `from_slice` being unused.
+
+## Step 5 (Host): Generate a receipt and read its journal contents
+
+For this step, we return to the main file for the host driver program at `factors/host/src/main.rs`, which currently has a placeholder comment asking to fill in with code for handling the [receipt]:
+
+```rust
+    // TODO: Implement code for transmitting or serializing the receipt for
+    // other parties to verify here
+```
+
+In a real-world scenario, we'd want to hand the [receipt] to someone else, but reading it ourselves will be a nice way to check our project is working as expected.
+So, let's extract the [journal]'s contents by replacing the "`TODO`" in the above code snippet with the following lines.
+
+```rust
+    // Extract journal of receipt (i.e. output c, where c = a * b)
+    let c: u64 = from_slice(&receipt.journal).unwrap();
+
+    // Print an assertion
+    println!("Hello, world! I know the factors of {}, and I can prove it!", c);
+```
+
+You should now be able to see your proof of factorization with `cargo run --release`.
+If your program printed the "Hello, world!" assertion and [receipt] verification was a success, congratulations!
+If not, we hope that troubleshooting will get you familiar with the system, and we'd love to chat with you on [Discord].
+Or, if you believe you've found a bug or other problem in our code, please open an [issue] describing the problem.
+
+If you're ready to start building more complex projects, we recommend taking a look at the other examples in our [examples directory] and looking through our further [Getting Started resources] for more project ideas that use zero-knowledge proofs.
 
-For a walk-through of this example, check out this [excerpt from our workshop at ZK HACK III](https://www.youtube.com/watch?v=nWxL21hKV9s&list=PLcPzhUaCxlCgig7ofeARMPwQ8vbuD6hC5&index=3).
-Note that this presentation was based on v0.11 of RISC Zero, and some API changes have been made since then.
+[install Rust]: https://www.rust-lang.org/tools/install
+[`cargo risczero` tool]: https://crates.io/crates/cargo-risczero
+[examples directory]: https://github.com/risc0/risc0/tree/main/examples
+[Getting Started resources]: https://www.risczero.com/docs/
+[Discord]: https://discord.com/invite/risczero
+[issue]: https://github.com/risc0/risc0/issues
```

Comment:
```suggestion
[issues]: https://github.com/risc0/risc0/issues
```

Reply:
no, this link must match the text or it won't link, and the text is singular:
> please open an [issue] describing the problem